tasks/init.js

```diff
@@ -35,83 +35,202 @@ var config = defaults.getInstance('init')
     editorconfig: true
   });
 
-var check = yargs.createCheck()
-  // don't check if we are just accessing help
-  .withGate(function (argv) {
-    return !argv.help;
-  })
-  .withTest({
-    name: function(value) {
-      if (typeof value !== 'string') {
-        return 'name must be a string';
-      } else if (value.length === 0) {
-        return 'name must not be zero length';
-      }
-    },
-    version: function(value) {
-      if (typeof value !== 'string') {
-        return 'version must be a valid string';
-      } else if (!/\d+\.\d+\.\d+[-\w\d]*/.test(value)) {
-        return 'version must be 3-term semver string';
+var initOptionDefinitions = [
+  {
+    key: 'help',
+    value: {
+      describe: 'This help message',
+      alias   : [ 'h', '?' ],
+      boolean : true
+    }
+  },
+  {
+    key: 'defaults',
+    value: {
+      describe: 'Set defaults',
+      alias   : 'z',
+      boolean : true,
+      isOptional: true
+    }
+  },
+  {
+    key: 'name',
+    value: {
+      describe: 'The project name',
+      alias   : 'n',
+      string  : true,
+      default : config.get('name')
+    }
+  },
+  {
+    key: 'version',
+    value: {
+      describe: 'The project version',
+      alias   : 'v',
+      string  : true,
+      default : config.get('version')
+    }
+  },
+  {
+    key: 'description',
+    value: {
+      describe: 'The project description',
+      alias   : 'd',
+      string  : true,
+      default : config.get('description')
+    }
+  },
+  {
+    key: 'tag',
+    value: {
+      describe: 'A project tag',
+      alias   : 't',
+      string  : true,
+      isMultiple: true, // expects an array
+      default : config.get('tag')
+    }
+  },
+  {
+    key: 'port',
+    value: {
+      describe: 'A port for the development web server',
+      alias   : 'p',
+      // boolean : true,
+      // string  : true,
+      default : config.get('port')
+    }
+  },
+  {
+    key: 'npm',
+    value: {
+      describe: 'Create package.json',
+      boolean : true,
+      default : config.get('npm')
+    }
+  },
+  {
+    key: 'bower',
+    value: {
+      describe: 'Create bower.json',
+      boolean : true,
+      default : config.get('bower')
+    }
+  },
+  {
+    key: 'karma',
+    value: {
+      describe: 'Create karma.conf.js',
+      boolean : true,
+      default : config.get('karma')
+    }
+  },
+  {
+    key: 'jshint',
+    value: {
+      describe: 'Create .jshintrc',
+      boolean : true,
+      default : config.get('jshint')
+    }
+  },
+  {
+    key: 'gitignore',
+    value: {
+      describe: 'Create .gitignore',
+      boolean : true,
+      default : config.get('gitignore')
+    }
+  },
+  {
+    key: 'editorconfig',
+    value: {
+      describe: 'Create .editorconfig',
+      boolean : true,
+      default : config.get('editorconfig')
+    }
+  },
+];
+
+function checkInitFlags(argv) {
```

Comment:
As you know but I will restate here.

This code is obtuse in this setting. The declarative part is the `initOptionDefinitions`, which is all the user wants to define. The code that processes it should be encapsulated in the third party library.